platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -433,17 +434,35 @@ OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
             {
                 /* POSIX port using standard library */
                 /* coverity[misra_c_2012_rule_21_6_violation] */
-                C->pFile = fopen( ( const char * ) realFilePath, "w+b" );
-
                 if( C->pFile != NULL )
```

Comment:
A comment would be good to mention about this logic closing a file if already open.

Reply:
Added in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1551/commits/b99d4ab575f455287a26210bb5d10cf318aa3d03